core/types.ml

```diff
@@ -131,7 +131,40 @@ type lens_phrase =
 
 (* End of Lenses *)
 
-type typ =
+type rec_id =
```

Comment:
Maybe add a line or two of documentation explaining the need for these two cases?